aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1281,6 +1308,118 @@ std::tuple<Tensor, Tensor> _symeig_helper_cuda(const Tensor& self, bool eigenvec
   }
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ eig ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
```

Comment:
Brief description of what this function does - and why it takes a CPU tensor.  Should this assert that self is on the CPU? 

Also, the name "apply_eig" is very generic. Is there a better, more CUDA-specific name for this function? "apply_eig_using_cuda"?

Reply:
>  why it takes a CPU tensor. Should this assert that self is on the CPU?

done by 6b5f0ce055. Is there a better way to do an assert, or `TORCH_CHECK` is fine?

> Also, the name "apply_eig" is very generic. Is there a better, more CUDA-specific name for this function? "apply_eig_using_cuda"?

maybe, but note that this is the naming convention of the whole file: every XXX operator has an `apply_XXX` function, so I think it's better to stick with it

Reply:
TORCH_CHECK is correct for assertions that are a result of user behavior, TORCH_INTERNAL_ASSERT is for assertions that are due to internal logical errors.